### PR TITLE
chore: Update DataFusion pin

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* Get fix for panic https://github.com/apache/arrow-datafusion/pull/6522 cc @NGA-TRAN
* Also keep up with dependencies and prepare for release next week

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/859251b4a20e00c3dfe73eee6b605fcf722687e7

### CODE REVIEW

File: influxdb_iox/tests/query_tests/cases/in/date_bin.sql.expected [hunk 1]
```diff
@@ -1,67 +1,67 @@
...
-+---------------------------------------------------------------------------+
-| 2022-01-01T00:00:00Z                                                      |
-| 2022-01-01T00:00:00Z                                                      |
-| 2022-01-01T00:00:00Z                                                      |
-| 2022-02-01T00:00:00Z                                                      |
-| 2022-02-01T00:00:00Z                                                      |
-| 2022-03-01T00:00:00Z                                                      |
-+---------------------------------------------------------------------------+
++----------------------------------------------------------------------------+
+| date_bin(IntervalMonthDayNano("79228162514264337593543950336"),sq.column1) |
```
Comment (alamb) [inline influxdb_iox/tests/query_tests/cases/in/date_bin.sql.expected:4]: The plan diffs are due to better display of function names https://github.com/apache/arrow-datafusion/pull/6448 which I think is a nice usability improvement

File: iox_query/src/physical_optimizer/predicate_pushdown.rs [hunk 2]
```diff
@@ -314,7 +314,7 @@ mod tests {
             projection: None,
             limit: None,
             table_partition_cols: vec![],
-            output_ordering: None,
+            output_ordering: vec![vec![]],
```
Comment (alamb) [inline iox_query/src/physical_optimizer/predicate_pushdown.rs:317]: The API changed to support multiple equivalent orderings: https://github.com/apache/arrow-datafusion/pull/6469

File: iox_query_influxql/src/plan/planner.rs [hunk 3]
```diff
@@ -2592,79 +2592,79 @@ mod test {
                 Projection: Dictionary(Int32, Utf8("all_types")) AS iox::measurement, tagKey [iox::measurement:Dictionary(Int32, Utf8), tagKey:Utf8;N]
                   Filter: tagKey IS NOT NULL [tagKey:Utf8;N]
                     Unnest: tagKey [tagKey:Utf8;N]
-                      Projection: makearray(CASE WHEN tag0 > Int32(0) THEN Utf8("tag0") END, CASE WHEN tag1 > Int32(0) THEN Utf8("tag1") END) AS tagKey [tagKey:FixedSizeList(Field { name: "item", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, 2);N]
+                      Projection: make_array(CASE WHEN tag0 > Int32(0) THEN Utf8("tag0") END, CASE WHEN tag1 > Int32(0) THEN Utf8("tag1") END) AS tagKey [tagKey:FixedSizeList(Field { name: "item", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, 2);N]
```
Comment (alamb) [inline iox_query_influxql/src/plan/planner.rs:2595]: The diffs in this file are also due to changes in how function names are rendered in plans

File: influxdb_iox/tests/query_tests/cases/in/duplicates_parquet.sql.expected [hunk 4]
```diff
@@ -96,16 +96,15 @@
...
-|    |           CoalesceBatchesExec: target_batch_size=8192, metrics=[elapsed_compute=1.234ms, output_rows=7]    |
-|    |             FilterExec: state@5 = MA, metrics=[elapsed_compute=1.234ms, output_rows=7]    |
-|    |               ParquetExec: file_groups={2 groups: [[1/1/1/00000000-0000-0000-0000-000000000002.parquet], [1/1/1/00000000-0000-0000-0000-000000000003.parquet]]}, projection=[__chunk_order, area, city, max_temp, min_temp, state, time], output_ordering=[state@5 ASC, city@2 ASC, time@6 ASC, __chunk_order@0 ASC], predicate=state@4 = MA, pruning_predicate=state_min@0 <= MA AND MA <= state_max@1, metrics=[bytes_scanned=1532, elapsed_compute=1.234ms, num_predicate_creation_errors=0, output_rows=7, page_index_eval_time=1.234ms, page_index_rows_filtered=0, predicate_evaluation_errors=0, pushdown_eval_time=1.234ms, pushdown_rows_filtered=3, row_groups_pruned=0, time_elapsed_opening=1.234ms, time_elapsed_processing=1.234ms, time_elapsed_scanning_total=1.234ms, time_elapsed_scanning_until_data=1.234ms]    |
+| Plan with Metrics    | UnionExec, metrics=[elapsed_compute=1.234ms, output_rows=10]    |
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply influxdb_iox/tests/query_tests/cases/in/duplicates_parquet.sql.expected:99]: Question: I see the UnionExec no longer in the new plan. Does it go with ` CoalescePartitionsExec is no longer added incorrectly`?

**Review by NGA-TRAN — APPROVED** [review]
Nice improvements in DF

**Pull request merged**